spi/control-plane/contract-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/contract/offer/ContractOffer.java

```diff
@@ -202,6 +202,10 @@ public Builder policy(Policy policy) {
         public ContractOffer build() {
             Objects.requireNonNull(id);
 
+            if (asset == null) {
```

Comment:
an alternative would be `Objects.requireNonNull(asset, "Asset must not be null")`, which would throw a NPE, but it would make the code a bit more readable, and we have it in many other places in the code base as well.

Reply:
Since i'm here should i change also the policy check?

Reply:
yes pls :)